ios/Classes/DarwinAudioSession.m

```diff
@@ -407,6 +411,28 @@ - (void) audioInterrupt:(NSNotification*)notification {
     }
 }
 
+- (void) routeChange:(NSNotification*)notification {
+    NSNumber *routeChangeReasonType = (NSNumber*)[notification.userInfo valueForKey:AVAudioSessionRouteChangeReasonKey];
+    NSLog(@"routeChange detected");
+    [self invokeMethod:@"onRouteChange" arguments:@[@([routeChangeReasonType integerValue])]];
+}
+
+- (void) silenceSecondaryAudio:(NSNotification*)notification {
+    NSNumber *silenceSecondaryType = (NSNumber*)[notification.userInfo valueForKey:AVAudioSessionSilenceSecondaryAudioHintTypeKey];
+    NSLog(@"silenceSecondaryAudioHint detected");
+    [self invokeMethod:@"onRouteChange" arguments:@[@([silenceSecondaryType integerValue])]];
```

Comment:
Is this the right method name?

Reply:
Definitely not, a copy paste error. Changed to "onSilenceSecondaryAudioHint".